### DISCUSSION TITLE
Is it possible to Scan a parameter space using PYHF?

### DISCUSSION THREAD

**Discussion by MohammadMohammadidoust** [scikit-hep/pyhf#2069] [Q&A]
Dear All
I am new to pyhf and also new to experimental HEP. I read the pyhf tutorial and have a question. As you mentioned in the section "Using Hep data", to investigate a parameter point, we should create a model using a patch from the given patchset and calculate CLs for this point, and then based on the CLs value we decide to exclude this point or not.
But my question is how to find out about the whole parameter space. the number of parameter points in patchsets is limited and it seems that we should use interpolation methods to find whether each arbitrary point of a parameter space should be excluded or not. could I do this using pyhf?
In other words, I need something like the one you have plotted in the tutorial(ATLAS results plot in the section "using HEPdata" of the tutorial) to flag each point in the parameter space regarding its CLs
can I achieve this using pyhf?

![plot_countour](https://user-images.githubusercontent.com/75882487/204740761-eaa6c838-21cf-4b0d-93cb-1c71bff6f756.gif)

**Comment by kratsg** [comment]
Hi, this is somewhat out of scope of `pyhf`. You are right in that you need to perform interpolation (what kind is really up to you; e.g. linear, quartic, ...). You can see what is done in HistFitter using [scipy interpolation](https://github.com/histfitter/histfitter/blob/master/scripts/harvestToContours.py#L537-L568) which relies on making a [`harvest.json`](https://github.com/histfitter/histfitter/blob/master/macros/Examples/contourPlotterExample/inputJSON/MySimpleChannelAnalysis_fixSigXSecNominal_hypotest__1_harvest_list.json). Once you have this TGraph however, you can make a plot like what's shown in your picture using this [gist](https://gist.github.com/kratsg/4ff8cb2ded3b25552ff2f51cd6b854dc) I made depending on matplotlib and [mplhep](https://github.com/scikit-hep/mplhep).
